core/src/main/java/com/novoda/noplayer/internal/exoplayer/NoPlayerAdsLoader.java

```diff
@@ -93,9 +93,18 @@ public void run() {
         }
 
         @Override
-        public void onAdvertsError(String message) {
+        public void onAdvertsError(final Exception cause) {
             loadingAds = null;
-            eventListener.onAdLoadError(null, null);
+            advertBreaks = Collections.emptyList();
+            adPlaybackState = new AdPlaybackState();
+
+            handler.post(new Runnable() {
+                @Override
+                public void run() {
+                    updateAdPlaybackState();
```

Comment:
`AdsMediaSource` doesn't load anything until a `AdPlaybackState` is passed to it. Even if adverts fail we need to pass this source in order for the content to play.